geopmctlpy/geopmctl/metric.py

```diff
@@ -0,0 +1,680 @@
+#  Copyright (c) 2015 - 2021, Intel Corporation
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions
+#  are met:
+#
+#      * Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in
+#        the documentation and/or other materials provided with the
+#        distribution.
+#
+#      * Neither the name of Intel Corporation nor the names of its
+#        contributors may be used to endorse or promote products derived
+#        from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+#  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+'''
+Defines Metric/Signal related classes and factory method(s).
+'''
+
+from enum import Enum
+from abc import ABC
+from abc import abstractmethod
+import yaml
+from geopmdpy import pio
+from geopmdpy import topo
+
+
+def create_signal(descriptor, report_formatter=None, trace_formatter=None):
+    '''
+    Factory function for creating a Signal object.
+
+    Args
+
+    descriptor (tuple of str, int, int): Tuple of name of the GEOPM signal, a domain enumeration
+    from the geopmdpy.topo package and instance number.
```

Comment:
```suggestion
    descriptor (tuple of str, int, int): Tuple of name of the GEOPM signal, a domain type and a domain index
    from the geopmdpy.topo package.
```

In all our other documentation, we refer to signals as a tuple of the signal name, domain type, and domain index.  Is this what you meant here?  

"Instance" is confusing.  Please use domain type and domain index if that's what you meant in all these comments.

Reply:
That's exactly what I mean, I will change the terminology I use.